### ISSUE TITLE
Feature Request: N1QL Support

### ISSUE THREAD

**Issue by jimhooker2002** [couchbase/CouchbaseMock#22]
I'm thinking about the following:

1: We can take a bit of inspiration from the design/structure/tooling of https://github.com/JSQLParser/JSqlParser to give us a "base" for this feature.

2: We can lift/reuse (and then tweak) the .ebnf/.y/.nex files from here: https://github.com/couchbase/query/ to give us the grammar we need.

Our mock implementation of the planner/executor would obviously be a very stripped down and simplified version.

In summary - lift the grammar from couchbase/query, then build much simplified versions of the planner/executor etc.

Thoughts?  Any other ideas around how we can get a N1QL support in the mock?

**Comment by mnunberg** [comment]
I think any solution which leaves us with at least a clean subset of N1QL should be good.

There's of course the challenge in properly managing data access.. executing "plans" and so on. I've mocked out some of the idea of plans though.

**Comment by unhuman** [comment]
+1 for this!

**Comment by tomrogers3** [comment]
+1

**Comment by msantosh1188** [comment]
We do looking fwd to have this feature, do let us know once you have things ready

**Comment by skycremaschinin** [comment]
hi everybody, i'm writing an application using couchbase and spring-data and i just know that i can't use this mock to test my app, spring data repositories uses n1ql and mock doesn't support it.
+100000000 for this!

**Comment by unhuman** [comment]
You could try testcontainers.org with couchbase.  Not perfect, but works.

**Comment by jurgob** [comment]
+1

**Comment by jurgob** [comment]
can we at least know if there's a plan to support this in couchbase Mock and what are the tech difficulties of doing this?

**Comment by himani1** [comment]
+1

**Comment by ZeeD** [comment]
+1

**Comment by yinan-liu** [comment]
+1

**Comment by EdgarLGB** [comment]
+1

**Comment by parag90** [comment]
+1 Any update from the Authors will be helpful If at all they are planning to build it.
